src/firmware/app/control/trajectory_planner.h

```diff
@@ -11,293 +13,434 @@ typedef struct FirmwareRobotPathParameters
- * @param forwards_continuous_velocity_profile [in/out] The forwards continuous velocity
- * profile to ebe modified in-place into a profile that is also backwards continuous
- * @param arc_segment_length [in] The arc segment length of each segment in the path the
- * profile is being generated for in meters
- * @param max_allowable_acceleration [in] The max allowable acceleration at any point on
- * the velocity profile
+ * @param speeds2 [in] The speed profile corresponding to trajectory 2. Speed must have a
+ * denominator value of seconds.
+ *
+ * @param durations1 [in] The time durations corresponding to the length of time between
+ * successive speed points in trajectory 1. In seconds.
+ *
+ * @param durations2 [in] The time durations corresponding to the length of time between
+ * successive speed points in trajectory 2. In seconds.
+ *
+ * @param displacement1 [in] The displacement corresponding to the distance between
+ * successive speed elements in trajectory 1. In meters.
+ *
+ * @param displacement2 [in] The displacement corresponding to the distance between
+ * successive speed elements in trajectory 2. In meters.
+ *
+ * @param complete_time_profile [out] The complete time profile of the two balanced
+ * trajectories. This profile is absolute time to each point on the profile starting from
+ * zero. In seconds.
+ *
+ * @param num_elements [in] The number of elements in both trajectory 1 and 2.
+ *
+ * @return A status indicating whether or not generation was successful
  */
-void app_trajectory_planner_generateBackwardsContinuousVelocityProfile(
-    const unsigned int num_segments,
-    float forwards_continuous_velocity_profile[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS],
-    const float arc_segment_length, const float max_allowable_acceleration);
+TrajectoryPlannerGenerationStatus_t app_trajectory_planner_modifySpeedsToMatchDuration(
+    float speeds1[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS],
+    float speeds2[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS],
+    float durations1[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS],
+    float durations2[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS],
+    float displacement1[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS],
+    float displacement2[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS],
+    float complete_time_profile[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS], float num_elements);
 
 /**
- *  Adds a time profile to an existing position profile for the given input parameters
+ * Returns a constant interpolation period (time) trajectory based on an input trajectory
  *
- *  @pre traj_elements must be pre-allocated and contain all of the position data of the
- * trajectory
- * @pre traj_elements & velocity_profile are pre-allocated up to at least
- * TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS
+ * This function is intended to take a variable time trajectory (which is easier to
+ * calculate) then interpolate a constant period trajectory. This is valuable for discrete
```

Comment:
could we choose one of `time` or `period` to use here for consistency? Similarly for the rest of this jdoc (and for jdocs here in general).

Reply:
Changed all references to constant-period